### PR TITLE
Support all log levels in the API and add user callback support

### PR DESCRIPTION
**Pull request by vdimir**

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review agent](https://github.com/flows-network/github-pr-review/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Comment by hydai** [comment]
Hi @vdimir 
Please sign your commits with DCO, thanks.
Ref: https://wasmedge.org/docs/contribute/#dco

**Comment by hydai** [comment]
The Windows CI failed because of an extra `;` in the `callback_sink.h`.
A fix is merged by the upstream: https://github.com/gabime/spdlog/pull/3198
We are unsure when will have a new release from the spdlog side. 
However, it's fine to disable this warning here to avoid the failing CI.

File: lib/api/wasmedge.cpp [hunk 1]
```diff
@@ -498,6 +498,51 @@ WASMEDGE_CAPI_EXPORT void WasmEdge_LogSetDebugLevel(void) {
...
+    WasmEdge::Log::setWarnLoggingLevel();
+    break;
+  case WasmEdge_LogLevel_Error:
+    WasmEdge::Log::setErrorLoggingLevel();
+    break;
+  case WasmEdge_LogLevel_Critical:
+    WasmEdge::Log::setCriticalLoggingLevel();
+    break;
+  default:
+    assumingUnreachable();
```
Comment (ibmibmibm) [inline lib/api/wasmedge.cpp:522]: This `Level` is provided by the user, so it cannot be assumed to always be valid.
You should check for invalid input here and decide whether to return an error or ignore it.

**Review by ibmibmibm — CHANGES_REQUESTED** [review]

**Comment by hydai** [comment]
@ibmibmibm 
PTAL. This PR should be ready to merge.

File: lib/api/wasmedge.cpp [hunk 2]
```diff
@@ -498,6 +498,49 @@ WASMEDGE_CAPI_EXPORT void WasmEdge_LogSetDebugLevel(void) {
...
+    break;
+  case WasmEdge_LogLevel_Critical:
+    WasmEdge::Log::setCriticalLoggingLevel();
+    break;
+  }
+}
+
+WASMEDGE_CAPI_EXPORT void
+WasmEdge_LogSetCallback(WasmEdge_LogCallback_t Callback) {
+  WasmEdge::Log::setLoggingCallback([Callback](
```
Comment (ibmibmibm) [inline lib/api/wasmedge.cpp:526]: Check `nullptr` and choose to fail or deregister callbacks

**Review by ibmibmibm — CHANGES_REQUESTED** [review]

**Comment by hydai** [comment]
Hi, @ibmibmibm 
I believe these are minor issues, and it's totally fine to fix them later via another PR. I would like to merge this PR first. Could you please approve it?

And thanks, @vdimir, for this; it helps a lot.

**Review by ibmibmibm — APPROVED** [review]

**Comment by hydai** [comment]
Thank you @vdimir

**Pull request merged**